stdlib/src/locale.rs

```diff
@@ -0,0 +1,154 @@
+        }
+        // https://github.com/python/cpython/blob/677320348728ce058fa3579017e985af74a236d4/Modules/_localemodule.c#L80
+        if !group_vec.is_empty() {
+            group_vec.push(vm.ctx.new_int(0i32).into());
+        }
+        vm.ctx.new_list(group_vec)
+    }
+
+    unsafe fn _parse_ptr_to_str(vm: &VirtualMachine, raw_ptr: *mut libc::c_char) -> PyResult {
+        let slice = unsafe { CStr::from_ptr(raw_ptr) };
+        let cstr = slice
+            .to_str()
+            .expect("localeconv always return decodable string");
+
+        Ok(vm.new_pyobj(cstr))
+    }
+
+    #[pyattr(name = "Error", once)]
+    fn error(vm: &VirtualMachine) -> PyTypeRef {
+        vm.ctx.new_exception_type(
+            "locale",
+            "Error",
+            Some(vec![vm.ctx.exceptions.exception_type.to_owned()]),
+        )
+    }
+
+    #[pyfunction]
+    fn localeconv(vm: &VirtualMachine) -> PyResult<PyDictRef> {
+        let result = vm.ctx.new_dict();
+
+        unsafe {
+            let lc = libc::localeconv();
+
+            macro_rules! set_string_field {
+                ($field:ident) => {{
+                    result.set_item(stringify!($field), _parse_ptr_to_str(vm, (*lc).$field)?, vm)?
+                }};
+            }
+
+            macro_rules! set_int_field {
+                ($field:ident) => {{
+                    result.set_item(stringify!($field), vm.new_pyobj((*lc).$field), vm)?
+                }};
+            }
+
+            macro_rules! set_group_field {
+                ($field:ident) => {{
+                    result.set_item(
+                        stringify!($field),
+                        copy_grouping((*lc).$field, vm).into(),
+                        vm,
+                    )?
+                }};
+            }
+
+            set_group_field!(mon_grouping);
+            set_group_field!(grouping);
+            set_int_field!(int_frac_digits);
+            set_int_field!(frac_digits);
+            set_int_field!(p_cs_precedes);
+            set_int_field!(p_sep_by_space);
+            set_int_field!(n_cs_precedes);
+            set_int_field!(p_sign_posn);
+            set_int_field!(n_sign_posn);
+            set_string_field!(decimal_point);
+            set_string_field!(thousands_sep);
+            set_string_field!(int_curr_symbol);
+            set_string_field!(currency_symbol);
+            set_string_field!(mon_decimal_point);
+            set_string_field!(mon_thousands_sep);
+            set_int_field!(n_sep_by_space);
+            set_string_field!(positive_sign);
+            set_string_field!(negative_sign);
+        }
+        Ok(result)
+    }
+
+    #[derive(FromArgs)]
+    struct LocaleArgs {
+        #[pyarg(any)]
+        category: i32,
+        #[pyarg(any, optional)]
+        locale: OptionalArg<Option<PyStrRef>>,
+    }
+
+    #[pyfunction]
+    fn setlocale(args: LocaleArgs, vm: &VirtualMachine) -> PyResult {
+        unsafe {
+            let result = match args.locale {
```

Comment:
this is handling both missing argument and `None` given same way, right?
using `flatten()` will merge two case and makes match simpler.
```suggestion
            let result = match args.locale.flatten() {
```